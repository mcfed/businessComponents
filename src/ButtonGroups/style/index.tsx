import 'antd/es/button/style';
import 'antd/es/tooltip/style';
import 'antd/es/menu/style';
import 'antd/es/icon/style';
import 'antd/es/dropdown/style';
import 'antd/es/modal/style';
