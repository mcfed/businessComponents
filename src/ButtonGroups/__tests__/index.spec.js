import React from 'react';
import {render, mount, shallow} from 'enzyme';
import ButtonGroups from '../index';

const Button = ButtonGroups.CustomButton;

const setup = props => {
  const wrapper = shallow(
    <ButtonGroups {...props} handleClick={jest.fn()}>
      <Button actionkey='aaa'>aaa</Button>
      <Button actionkey='bbb'>bbb</Button>
    </ButtonGroups>
  );
  return {
    wrapper
  };
};

describe.skip('ts-button-group test ', () => {
  it('cust button base test', () => {
    const wrapper = mount(
      <Button
        actionkey='aaa'
        tip='bbb'
        confirm='ccc'
        confirmTitle='ddd'
        permission={true}>
        test
      </Button>
    );

    expect(wrapper.find('.ant-btn').prop('disabled')).toEqual(undefined);
    expect(wrapper.find('.ant-btn').prop('type')).toEqual(undefined);
  });

  it('button groups base test mode buttngroups', () => {
    const wrapper = render(
      <ButtonGroups handleClick={() => {}}>
        <Button actionkey='aaaa'>aaa</Button>
        <Button actionkey='bbb'>bbb</Button>
        <Button actionkey='ccc' confirm='ccc'>
          ccc
        </Button>
        <Button actionkey='ccc' confirm='ccc' permission={false}>
          ccc
        </Button>
      </ButtonGroups>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('button groups base test mode ButtonMenu', () => {
    const wrapper = render(
      <ButtonGroups handleClick={() => {}} mode='ButtonMenu'>
        <Button actionkey='aaaa'>aaa</Button>
        <Button actionkey='bbb'>bbb</Button>
        <Button actionkey='ccc' confirm='ccc'>
          ccc
        </Button>
        <Button actionkey='ccc' confirm='ccc' permission={false}>
          ccc
        </Button>
      </ButtonGroups>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('button filter children method test', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const childArray = [
      <div permission={undefined}>1</div>,
      <div>2</div>,
      <div permission={true}>3</div>,
      <div permission={false}>4</div>
    ];
    const resultArray = [
      <div permission={undefined}>1</div>,
      <div>2</div>,
      <div permission={true}>3</div>
    ];

    expect(instance.filterChildren(childArray)).toEqual(resultArray);
  });

  it('method test formatTooltipTitle', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();

    expect(instance.formatTooltipTitle({tip: 'a', children: 'b'})).toEqual('a');
    expect(instance.formatTooltipTitle({tip: 'a'})).toEqual('a');
    expect(instance.formatTooltipTitle({children: 'b'})).toEqual('b');
  });
  it('method test renderReactElement tip1', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();

    instance.renderConfirmChild = jest.fn();
    instance.renderReactElement({props: {disabled: false, confirm: true}});
    expect(instance.renderConfirmChild).toHaveBeenCalled();
  });

  it('method test renderReactElement tip2', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();

    instance.renderNormalChild = jest.fn();
    instance.renderReactElement({props: {disabled: false, confirm: false}});
    expect(instance.renderNormalChild).toHaveBeenCalled();
  });

  it('method test renderConfirmChild ', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const renderProps = {
      props: {
        tip: 'aaa',
        confirm: 'bbb',
        actionkey: 'ccc',
        confirmTitle: 'ddd'
      }
    };

    const confirmChild = instance.renderConfirmChild(renderProps, 1);
    expect(confirmChild.props.content).toBe(renderProps.props.confirm);
    expect(confirmChild.props.title).toBe(renderProps.props.confirmTitle);
  });

  it('method test renderMenuChild', () => {
    const {wrapper} = setup();
    const instance = wrapper.instance();
    const renderProps = {
      props: {
        tip: 'aaa'
      }
    };
    const menuChild = instance.renderMenuChild(renderProps, 1);
    expect(menuChild.props.title).toEqual(renderProps.props.tip);
    expect(menuChild.key).toEqual('1');
  });
});
