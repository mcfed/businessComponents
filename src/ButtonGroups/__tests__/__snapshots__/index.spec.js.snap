// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-button-group test  button groups base test mode ButtonMenu 1`] = `
<div
  class="button-groups"
>
  <div
    class="ant-btn-group"
  >
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        aaa
      </span>
    </button>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        bbb
      </span>
    </button>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        ccc
      </span>
    </button>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        ccc
      </span>
    </button>
    <button
      class="ant-btn ant-dropdown-trigger"
      type="button"
    >
      <i
        aria-label="icon: ellipsis"
        class="anticon anticon-ellipsis"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="ellipsis"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M176 511a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0zm280 0a56 56 0 1 0 112 0 56 56 0 1 0-112 0z"
          />
        </svg>
      </i>
    </button>
  </div>
</div>
`;

exports[`ts-button-group test  button groups base test mode buttngroups 1`] = `
<div
  class="button-groups"
>
  <div
    class="ant-btn-group"
  >
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        aaa
      </span>
    </button>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        bbb
      </span>
    </button>
    <button
      class="ant-btn"
      type="button"
    >
      <span>
        ccc
      </span>
    </button>
  </div>
</div>
`;
